tool/tool-tool/su/src/java/org/sakaiproject/tool/su/SuTool.java

```diff
@@ -230,8 +232,15 @@ else if (users.size() > 1)
 				M_log.error(e.getMessage(), e);
 			}
 		}
-		
-		return "redirect";
+
+		//as user is authorised redirect back to portal instead of going to admin workspace
+		ExternalContext context = FacesContext.getCurrentInstance().getExternalContext();
+		try {
+			context.redirect(getPortalUrl());
+		} catch (IOException e) {
+			M_log.error("Failed to redirect to portal : " + e.getMessage());
```

Comment:
Can you save a message in JSF's state so that the user sees something went wrong.